src/rgw/rgw_bucket.cc

```diff
@@ -1585,8 +1585,15 @@ int RGWBucketAdminOp::limit_check(rgw::sal::RGWRadosStore *store,
 int RGWBucketAdminOp::info(rgw::sal::RGWRadosStore *store, RGWBucketAdminOpState& op_state,
                   RGWFormatterFlusher& flusher)
 {
+  RGWBucket bucket;
   int ret = 0;
   string bucket_name = op_state.get_bucket_name();
+  if (!bucket_name.empty()) {
```

Comment:
why don't we raise an error below when we call `bucket_stats()`?
```c++
  } else if (!bucket_name.empty()) {
    ret = bucket_stats(store, user_id.tenant, bucket_name, formatter);
    if (ret < 0) {
      return ret;
    }
```
the call to `get_bucket_info()` in `bucket_stats()` should be generating the same error

Reply:
It does generate an error, and it works fine for the cli, the issue is that we've already started writing the response for http api here: https://github.com/ceph/ceph/pull/34866/files#diff-684ce351ffd8f3eed0807d03ae52ca2fL1591
flusher.start(0);flusher.start(0)

This means we can't update the http return code after that point. If an error is returned after `flusher.start(0);`, the endpoint returns status `200` with an empty response body, which is not optimal.  This is a pretty widespread pattern for the admin api endpoints which limits the ways input validation can be done (BucketInfo misses a lot of cases with invalid users/buckets right now). Out of band to this change, I'm working on a little rewrite of this endpoint with its own `::send_message()` function that would separate data collection/validation from writing the http response.

Reply:
ref: https://github.com/ceph/ceph/blob/master/src/rgw/rgw_rest.cc#L1637-L1645

This is the method that I'd be overriding. Right now it's generically used by all the admin endpoints afaict.

Reply:
ok thanks. i remember looking into this before, and getting the Flusher logic right in all of those cases looked too complicated. i'm fine with your approach that adds an extra read of the bucket info - when `bucket_stats()` goes to read it again, it should at least be in the metadata cache